BirtherDay/Presentation/Home/HomeView.swift

```diff
@@ -25,38 +25,11 @@ struct HomeView: View {
                     unusedCouponListView()
                 }
                 .navigationDestination(for: BDAppPath.self) { path in
-                    switch path {
-                    case .create(let createPath):
-                        switch createPath {
-                        case .selectTemplate:
-                            CouponTemplateView(viewModel: couponViewModel)
-                        case .couponInfo:
-                            CouponInfoView(viewModel: couponViewModel)
-                        case .couponLetter:
-                            CouponLetterView(viewModel: couponViewModel)
-                        case .couponPicture:
-                            CouponPhotoView(viewModel: couponViewModel)
-                        case .couponComplete:
-                            CouponCompleteView()
-                        }
-                    case .myCoupon(let myPath):
-                        switch myPath {
-                        case .couponInventory:
-                            MyCouponView()
-                        case .couponDetail:
-                            Text("CouponDetailView")
-                            CouponDetailView(viewModel: CouponDetailViewModel())
-                        case .interaction:
-                            CouponInteractionView()
-                        case .interactionComplete:
-                            InteractionCompleteView()
-                        }
-                    case .test(let testPath):
-                        switch testPath {
-                        case .test:
-                            TestView()
-                        }
-                    }
+                    BDNavigationRoutingView(
+                        destination: path,
+                        createCouponViewModel: couponViewModel,
+                        couponDetailViewModel: CouponDetailViewModel()
```

Comment:
p1

인라인 생성이 아닌, couponViewModel처럼 외부에서 생성한 이후에 주입하는 것이 일관성있고 가독성이 좋은 것 같아요~~

```suggestion
                       private let couponDetailViewModel = CouponDetailViewModel()
//                       ...
                        couponDetailViewModel: couponDetailViewModel
```

Reply:
아 이해했어요 근데 내비게이팅뷰가 HomeView에서 연결되는 뷰라 외부에서 뷰모델을 선언하고 주입하려면, HomeView에 CouponDetailViewModel()를 초기화해야 합니다!! 나중에 내비게이션 라우팅 뷰 대대적으로 수정들어갈게요...